src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -45,8 +45,11 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 hostBuilder.UseSetting(nameof(UseIISIntegration), "true");
                 hostBuilder.CaptureStartupErrors(true);
 
-                var applicationPath = NativeMethods.http_get_application_full_path();
-                hostBuilder.UseContentRoot(applicationPath);
+                // TODO consider adding a configuration load where all variables needed are loaded from ANCM in one call.
+                var hResult = NativeMethods.http_get_application_paths(out var fullPath, out var virtualPath);
+                IISApplicationConfiguration.SetVirtualPath(virtualPath);
```

Comment:
Yea, nah. This can be an part of the options that flows into the `IISHttpServer` implementation.